__tests__/app.test.js

```diff
@@ -161,3 +161,65 @@ test('GET: 200 sends an empty array if article exists but has no comments', () =
     expect(body.comments).toEqual([])
   })
 })
+test('POST: 201 adds a new comment to the article which is an object', () => {
+  const newComment = {
+    body: 'my new comment',
+    username: 'icellusedkars'}
+  return request(app)
+  .post("/api/articles/13/comments")
+  .send(newComment)
+  .expect(201)
+  .then(({ body }) => {
+    expect(body.comment).toMatchObject({body: 'my new comment', 
+      comment_id: expect.any(Number),
+      article_id: 13,
+      author: 'icellusedkars',
+      votes: 0,
+      created_at: expect.any(String)})
+
+})
+})
+test('POST:400 responds with an appropriate status and error message when provided with incomplete data)', () => {
+  const newComment = {body: 'my new comment'}
+  return request(app)
+    .post('/api/articles/13/comments')
+    .send(newComment)
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe('Bad request');
+    });
+});
+test('POST:404 sends an appropriate status and error message when given a valid but non-existent id', () => {
+  const newComment = {
+    body: 'my new comment',
+    username: 'icellusedkars'}
+  return request(app)
+    .post('/api/articles/999/comments')
+    .send(newComment)
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe('article does not exist');
+    });
+})
+test('POST:400 responds with an appropriate status and error message when provided with incorrect username)', () => {
```

Comment:
This should be a 404 really.
It is a good request but the username cannot be found.